src/test/java/com/ksm/wstbackoffice/controller/CountryControllerIT.java

```diff
@@ -54,6 +71,52 @@ public void findAllTest() {
             get("countries").
         then().
             statusCode(200).
-            body("size()", equalTo(5));
+            body("size()", equalTo(21));
+    }
+
+    @ParameterizedTest
+    @MethodSource("findCountriesStartWithTestArguments")
+    public void findCountriesStartWithTest(List<String> nameStartsWithFilters, int expectedStatus) {
+        given().
+            contentType(ContentType.JSON).
+            body(nameStartsWithFilters).
+        when().
+            post("countries/filter-name").
+        then().
+            statusCode(expectedStatus);
+    }
+
+    @Test
+    public void findCountriesStartWithTest() {
+        List<String> nameStartsWithFilters = Arrays.asList("sa");
+
+        given().
+            contentType(ContentType.JSON).
+            body(nameStartsWithFilters).
+        when().
+            post("countries/filter-name").
+        then().
+            statusCode(200)
+            .body("details.sa", hasItems("Saint Kitts and Nevis", "Saint Lucia", "Saint Vincent and the Grenadines"))
```

Comment:
Is it possible to specify these values in the test arguments provider?

Reply:
done